docs/collecting-data/collecting-from-own-applications/python-tracker/subject/index.md

```diff
@@ -0,0 +1,236 @@
+
+### Change the tracker's platform with `set_platform`
+
+The default platform is "pc". You can change the platform the subject is using by calling for example:
+
+```python
+s.set_platform('mob')
+```
+
+For a full list of supported platforms, please see the [Snowplow Tracker Protocol](/docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/index.md#application-parameters).
+
+### Set user ID with `set_user_id`
+
+You can set the user ID to any string:
+
+```python
+s.set_user_id("user_id")
+```
+
+### Set screen resolution with `set_screen_resolution`
+
+If your Python code has access to the device's screen resolution, then you can pass this in to Snowplow too. 
+Both numbers should be positive integers; note the order is width followed by height. For example:
+
+```python
+s.set_screen_resolution(1366, 768)
+```
+
+### Set viewport dimensions with `set_viewport`
+
+If your Python code has access to the viewport dimensions, then you can pass this in to the subject too. Both numbers should be positive integers; note the order is width followed by height. For example:
+
+```python
+s.set_viewport(300, 200)
+```
+
+### Set color depth with `set_color_depth`
+
+If your Python code has access to the bit depth of the device's color palette for displaying images, then you can pass this in to the subject too. The number should be a positive integer, in bits per pixel. For example:
+
+```python
+s.set_color_depth(32)
+```
+
+### Set timezone with `set_timezone`
+
+This method lets you pass a user's timezone into Snowplow. The timezone should be a string.
+
+```python
+s.set_timezone("Europe/London")
+```
+
+### Set the language with `set_lang`
+
+This method lets you pass a user's language into Snowplow. The language should be a string.
+
+```python
+s.set_lang('en')
+```
+
+### Setting the IP address with `set_ip_address`
+
+If you have access to the user's IP address, you can set it like this:
+
+```python
+s.set_ip_address('34.633.11.139')
+```
+
+### Setting the useragent with `set_useragent`
+
+If you have access to the user's useragent (sometimes called "browser string"), you can set it like this:
+
+```python
+s.set_useragent('Mozilla/5.0 (Windows NT 5.1; rv:23.0) Gecko/20100101 Firefox/23.0')
+```
+
+### Setting the domain user ID with `set_domain_user_id`
+
+The `domain_userid` field of the Snowplow event model corresponds to the ID stored in the first party cookie set by the Snowplow JavaScript Tracker. If you want to match up server-side events with client-side events, you can set the domain user ID for server-side events like this:
+
+```python
+s.set_domain_user_id('c7aadf5c60a5dff9')
+```
+
+You can extract the domain user ID from the cookies of a request using the `get_domain_user_id` function below. The `request` argument is, as an example, the [Django request object](https://docs.djangoproject.com/en/1.7/ref/request-response/).
+
+**Note that this function has not been tested.**
+
```

Comment:
can we test it?

Reply:
I'm just going to bin it off, not sure we really need django docs in here.

Reply:
unchained, but i'll allow it